Fix pkg/ depends on cmd/ problems

dims opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
-->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Partial fix for https://github.com/kubernetes/kubernetes/issues/53341

**Special notes for your reviewer**:
No logic changes, Just moving things around

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
NONE
```

dims committed:
Make sure pkg/controller/service does not depend on cmd/kubeadm/app

dims committed:
Remove cmd/kubeadm from pkg/controller

dims committed:
Remove cmd/kubelet dependency from pkg/kubelet/volumemanager

dims committed:
Remove cmd/kubelet dependency from pkg/kubelet

dims commented:
Down to a handful
```
[dims@dims-mac 15:37] ~/go/src/k8s.io/kubernetes ⟩ find pkg/ -name BUILD -exec grep -H -i -E "(//|//.*/)cmd" {} \; | grep -v pkg/kubectl/cmd
pkg/api/testing/BUILD:        "//cmd/kubeadm/app/apis/kubeadm/fuzzer:go_default_library",
pkg/controller/certificates/BUILD:        "//cmd/gke-certificates-controller:__subpackages__",
pkg/kubemark/BUILD:        "//cmd/kube-proxy/app:go_default_library",
pkg/kubemark/BUILD:        "//cmd/kubelet/app:go_default_library",
pkg/kubemark/BUILD:        "//cmd/kubelet/app/options:go_default_library",
```

dims commented:
- the `pkg/controller/certificates/BUILD`  will go away when https://github.com/kubernetes/kubernetes/issues/53439 gets fixed
- the `pkg/api/testing/BUILD` will go away when kubeadm is moved to a separate repo

Not sure if we want to touch `pkg/kubemark`

dims committed:
Fix license boilerplate

dims committed:
Drop cmd/gke-certificates-controller from bazel build script

dims commented:
/retest

cblecker reviewed (approved):
👍

mikedanese commented:
Can you add an import boss config so this dep doesn't happen again?

dims commented:
@mikedanese - can you please point me to some thing? not sure what `import boss config` means

cblecker commented:
I think he's talking about:
https://github.com/kubernetes/kubernetes/blob/41f928f93050e733f9dd4ff09e418bb1fb44b0be/staging/src/k8s.io/code-generator/cmd/import-boss/main.go#L17-L29
which is run as a verify step (`hack/verify-import-boss.sh`)

dims commented:
Ah! thanks @cblecker will do that

dims committed:
Add import-boss directives

The pkg/api/testing can be removed whem kubeadm splits into
its own repository.
The pkg/kubemark probably will remain as is for the foreseeable
future.

dims committed:
Ignore .import-restrictions when checking against generated docs

dims commented:
/retest

cblecker commented on hack/verify-generated-swagger-docs.sh:
```diff
@@ -55,7 +55,7 @@ cp -a "${DIFFROOT}"/* "${TMP_DIFFROOT}/"
 echo "diffing ${DIFFROOT} against freshly generated swagger type documentation"
 ret=0
-diff -Naupr -I 'Auto generated by' "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
+diff --exclude=".import-restrictions" -Naupr -I 'Auto generated by' "${DIFFROOT}" "${TMP_DIFFROOT}" || ret=$?
```
If you regenerate the docs completely, does the file get wiped out?

dims replied:
y, running `hack/update-generated-swagger-docs.sh` seems to get rid of the file

dims replied:
so `--exclude` may not be a great idea :)

cblecker replied:
yeah, that's my thoughts. not sure how to handle the import-restrictions file then =\

dims replied:
Hmm, sorry i checked again, the generated files in _tmp/pkg does not have the file, We don't wipe the existing file from disk. So this change is still legit

```
Generating swagger type docs for settings.k8s.io/v1alpha1 at vendor/k8s.io/api/settings/v1alpha1
Generating swagger type docs for storage.k8s.io/v1beta1 at vendor/k8s.io/api/storage/v1beta1
Generating swagger type docs for storage.k8s.io/v1 at vendor/k8s.io/api/storage/v1
diffing /Users/dims/go/src/k8s.io/kubernetes/pkg against freshly generated swagger type documentation
diff -Naupr -I 'Auto generated by' /Users/dims/go/src/k8s.io/kubernetes/pkg/.import-restrictions /Users/dims/go/src/k8s.io/kubernetes/_tmp/pkg/.import-restrictions
--- /Users/dims/go/src/k8s.io/kubernetes/pkg/.import-restrictions	2017-10-07 21:36:12.000000000 -0400
+++ /Users/dims/go/src/k8s.io/kubernetes/_tmp/pkg/.import-restrictions	1969-12-31 19:00:00.000000000 -0500
@@ -1,13 +0,0 @@
-{
-	"Rules": [
-	  {
-		"SelectorRegexp": "k8s[.]io/kubernetes",
-		"AllowedPrefixes": [
-		    ""
-		],
-		"ForbiddenPrefixes": [
-			"k8s.io/kubernetes/cmd"
-		]
-	  }
-	]
-}
/Users/dims/go/src/k8s.io/kubernetes/pkg is out of date. Please run hack/update-generated-swagger-docs.sh
```

dims commented:
/assign @thockin

dims commented:
@thockin - we can ignore the failing test it's unrelated infra issue possibly (https://github.com/kubernetes/kubernetes/issues/53570)

thockin commented on pkg/master/master.go:
```diff
@@ -265,7 +265,7 @@ func (cfg *Config) Complete(informers informers.SharedInformerFactory) Completed
 		// but then that breaks the strict nestedness of ServiceType.
 		// Review post-v1
-		c.ExtraConfig.ServiceNodePortRange = options.DefaultServiceNodePortRange
+		c.ExtraConfig.ServiceNodePortRange = kubeoptions.DefaultServiceNodePortRange
```
Uggggggh.  All of this default-setting in multiple places is horrific.  It's been a long time since I touched this, but could we find a way to force callers to pass valid values?

dims replied:
@thockin yes we can, in a follow up PR. Want to keep it as simple as i can.

thockin commented on cmd/kubelet/app/options/container_runtime.go:
```diff
@@ -39,68 +38,15 @@ var (
 		defaultPodSandboxImageVersion
 )
 
-type ContainerRuntimeOptions struct {
```
@mtaufen

dims replied:
Just moved things around, No logic change.

thockin commented on pkg/controller/certificates/BUILD:
```diff
@@ -10,10 +10,7 @@ go_library(
         "certificate_controller.go",
         "certificate_controller_utils.go",
     ],
-    visibility = [
```
I don't understand this one?

dims replied:
Does not seem like it's needed. verify bazel does not complain. So i just removed it.

tallclair replied:
visibility is the bazel equivalent of import-restrictions.

thockin replied:
so, we shouldn't be removing these?

dims commented:
/test pull-kubernetes-unit

dims commented:
/test pull-kubernetes-unit

dims commented:
/assign @bgrant0607 

Brian, Tim, it's getting hard to keep this up-to-date. PTAL. I really like the import-boss restriction to make sure we don't leak any more stuff.

tallclair commented on pkg/controller/service/service_controller.go:
```diff
@@ -63,6 +62,14 @@ const (
+
+	// LabelNodeRoleExcludeBalancer specifies that the node should be
+	// exclude from load balancers created by a cloud provider.
+	LabelNodeRoleExcludeBalancer = "node.role.kubernetes.io/exclude-balancer"
```
Seems like these should be part of the API? A labels equivalent of https://github.com/kubernetes/kubernetes/blob/fc1bd98c09e2a91e9926b289ceede6593f96fe86/staging/src/k8s.io/api/core/v1/annotation_key_constants.go

cblecker replied:
I think this needs to be updated with the change from #53678

dims replied:
Good catch! Thanks @cblecker

thockin commented:
/lgtm
/approve

luxas commented on pkg/controller/service/service_controller.go:
```diff
@@ -63,6 +62,14 @@ const (
+
+	// LabelNodeRoleMaster specifies that a node is a master
+	// It's copied over to kubeadm until it's merged in core: https://github.com/kubernetes/kubernetes/pull/39112
+	LabelNodeRoleMaster = "node-role.kubernetes.io/master"
```
I don't think this label should live here. Please move it back as a constant (duplicated until we have a "real" place for it) to `cmd/kubeadm/app/constants` _as well_. I don't want `cmd/kubeadm` to depend on `pkg/controller/service` just for this.

@dims OK for you? Can you please fix up?

dims replied:
@luxas Done. Please see https://github.com/kubernetes/kubernetes/pull/53937
